scenes/game/local_game.gd

```diff
@@ -2593,7 +2629,20 @@ func handle_strike_effect(card_id :int, effect, performing_player : Player):
 			events += [create_event(Enums.EventType.EventType_Strike_DodgeFromOppositeBuddy, performing_player.my_id, 0, "", effect['buddy_name'])]
 			_append_log("%s is now dodging attacks from opponents behind %s." % [performing_player.name, effect['buddy_name']])
 		"draw":
-			events += performing_player.draw(effect['amount'])
+			var draw_amount = effect['amount']
+			var deck_size = len(performing_player.deck)
+			if draw_amount <= deck_size:
+				events += performing_player.draw(draw_amount)
+			else:
+				# Draw what you can now
+				var remainder = draw_amount - deck_size
+				events += performing_player.draw(deck_size)
+
+				# Process reshuffle, and prepare the remainder of the draw for after
+				var followup_effect = effect.duplicate()
+				followup_effect['card_id'] = card_id
+				followup_effect['amount'] = remainder
+				events += performing_player.begin_reshuffle(false, followup_effect)
```

Comment:
I really, really don't like adding a decision inside draw, as basically in all cards created so far, the effects were written on the assumption that draw is not a decision point.
I also don't like moving the reshuffling logic outside of player.draw.
I don't want to have a blocking operation on the current player's turn for something as simple as drawing.

Instead of doing it this way, can we just send an event on reshuffle that provides the list of unknown cards somehow that then updates revealed cards?
Additionally, you could add something that marks up the deck reference somehow.

I'm just really worried this will break some obscure cases and make it more complicated to write all effects knowing draw is now a decision point.

Reply:
As mentioned in DMs it's probably better to go with an approach like this; I do think this branch is functional so I'll keep it around personally, but I'll probably close this PR and open a new one once I have that version done